contracts/token-distribution/SaleRounds.sol

```diff
@@ -204,115 +159,186 @@ contract SaleRounds is TokenDistribution, GameOwner, ERC20 {
+        RoundType roundType = getRoundTypeByKey(_roundType);
+
+        return reservedBalances[roundType][_to];
     }
 
-    function getTotalClaimedForAllRounds() public view returns(uint256) {
-       return totalSupply();
+    function getCliffTime(string calldata _roundType) external view onlyGameOwner returns(uint256) {
+        RoundType roundType =  getRoundTypeByKey(_roundType);
+
+        return roundDistribution[roundType].cliff;
     }
 
-    function getTotalRemainingForAllRounds() public view returns(uint256) {
-       (, uint val) = maxSupply.trySub(totalSupply());
-       return val;
+    function mintTokensForExchanges(address _to, uint _amount) public onlyOwner {
+        RoundType roundType = RoundType.EXCHANGES;
+        require(reservedBalances[roundType][_to] >= _amount, "amount is grater then total reserved balance");
+
+        ClaimInfo memory exchangesClaimInfo = ClaimInfo({
+        cliff: roundDistribution[roundType].cliff,
+        vesting: roundDistribution[roundType].vesting,
+        balance: reservedBalances[roundType][_to],
+        claimedBalance: claimedBalances[roundType][_to],
+        periodGranularity: roundDistribution[roundType].vestingGranularity,
+        startTime: roundDistribution[roundType].startTime,
+        secondsVested: 0,
+        vestingForUserPerSecond: 0
+        });
+
+        require(exchangesClaimInfo.balance > 0, "don't have a reserved balance");
+        exchangesClaimInfo.secondsVested = calculateCliffTimeDiff(exchangesClaimInfo);
+        require(exchangesClaimInfo.secondsVested > 0, string.concat("Exchanges round cliff time didn't expired"));
+        exchangesClaimInfo.vestingForUserPerSecond = calculateVestingForUserPerSecond(exchangesClaimInfo);
+
+        uint maximumRelease = getMaximumRelease(exchangesClaimInfo);
+
+        (uint balanceToRelease, uint unClaimedBalance) = getBalanceToRelease(maximumRelease, exchangesClaimInfo);
+
+        if (exchangesClaimInfo.vestingForUserPerSecond == 0) {
+            balanceToRelease = unClaimedBalance;
+        }
+        _mint(_to, balanceToRelease);
     }
 
-    function getTotalRemainingForSpecificRound(string calldata _roundType) public view returns(uint256) {
-        RoundType roundType = getRoundTypeByKey(_roundType);
-        return roundDistribution[roundType].totalRemaining;
+    // @_amount is going be decimals() == default(18) digits
+    function reserveTokensInternal(RoundType _roundType, address _to, uint _amount) private {
+        require(roundDistribution[_roundType].supply >= _amount, "given amount is bigger than max supply for the round");
+        require(roundDistribution[_roundType].totalRemaining >= _amount, "total remaining round amount is not enough");
+        roundDistribution[_roundType].totalRemaining -= _amount;
+        roundDistribution[_roundType].startTime = block.timestamp;
+        reservedBalances[_roundType][_to] += _amount;
```

Comment:
What happens when [_to] is not in the mapping yet, is it guaranteed to be 0?
Is this covered by tests?